src/types/coefficient.rs

```diff
@@ -20,8 +20,8 @@ pub enum Sign {
 // [Magnitude].
 #[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq)]
 pub struct Coefficient {
-    sign: Sign,
-    magnitude: Magnitude,
+    pub(crate) sign: Sign,
+    pub(crate) magnitude: Magnitude,
```

Comment:
Same as with timestamps, we should probably think about defining the encoding over traits so that we can do zerocopy encoding of BigDecimal.

Reply:
Just to clarify, I think you mean the "number"-ness for our decimal and `bigdecimal::BigDecimal`, yes?  In which case I very much agree.

Reply:
Added #286 to track this.